crates/proto/src/rr/dns_class.rs

```diff
@@ -48,7 +48,7 @@ impl FromStr for DNSClass {
     /// assert_eq!(DNSClass::IN, var);
     /// ```
     fn from_str(str: &str) -> ProtoResult<Self> {
-        match str {
+        match str.to_uppercase().as_str() {
```

Comment:
two things. I think this should be: https://doc.rust-lang.org/std/primitive.str.html#method.to_ascii_uppercase

There's no need to support UTF-8 here. Second, since this method takes an `&str` we should try not to allocate here...

So should we instead call https://doc.rust-lang.org/std/primitive.str.html#method.make_ascii_uppercase (which does an in place upppercase since it's ascii only), and do it here instead: https://github.com/bluejekyll/trust-dns/blob/main/crates/client/src/serialize/txt/zone.rs#L218

Then here if we want to make sure this isn't used incorrectly, we could do a `debug_assert!(str.chars().all(char::is_ascii_uppercase))`.

What do you think?

Reply:
Makes total sense.